### PR TITLE
Added special case for m.room.member event in appservice component

### LINKED ISSUE #648: We should check if application services are interested in a membership event's target
We should special case the processing of `m.room.member` events in the appservice component to call `appservice.IsInterestedInUserID` with the event's state key on top of the existing call with the event's sender.

### PR DESCRIPTION
**Pull request by prateek2211**
### Pull Request Checklist

* [ ] I have added any new tests that need to pass to `testfile` as specified in [docs/sytest.md](https://github.com/matrix-org/dendrite/blob/master/docs/sytest.md)
* [X] Pull request includes a [sign off](https://github.com/matrix-org/dendrite/blob/master/CONTRIBUTING.md#sign-off)

Signed-off-by: Prateek Sachan psachan@cs.iitr.ac.in
fixed #648

### CODE REVIEW

File: appservice/consumers/roomserver.go [hunk 1]
```diff
@@ -191,6 +191,12 @@ func (s *OutputRoomEventConsumer) appserviceIsInterestedInEvent(ctx context.Cont
 		return true
 	}
 
+	if event.Type() == "m.room.member" {
```
Comment (Kegsay) [inline appservice/consumers/roomserver.go]: Please use `gomatrixserverlib.MRoomMember` as the constant

Comment (Kegsay) [reply appservice/consumers/roomserver.go]: Also, please do a nil check on `StateKey` e.g:
```
if event.Type() == gomatrixserverlib.MRoomMember && event.StateKey() != nil {
```

**Comment by Kegsay** [comment]
Otherwise LGTM

File: appservice/consumers/roomserver.go (hunk 1 above)
Comment (prateek2211) [reply appservice/consumers/roomserver.go]: Done!

**Pull request merged**